docs/src/modules/components/AppLayoutDocsFooter.js

```diff
@@ -135,33 +142,53 @@ function getCurrentRating(pathname) {
   return userFeedback && userFeedback[pathname] && userFeedback[pathname].rating;
 }
 
+/**
+ * @returns { { prevPage: OrderedMuiPage | null; nextPage: OrderedMuiPage | null } }
+ */
+function usePageNeighbours() {
+  const { activePage, pages } = React.useContext(PageContext);
+  const pageList = orderedPages(pages);
+  const currentPageNum = pageList.indexOf(activePage);
+  if (currentPageNum === -1) {
+    return { prevPage: undefined, nextPage: undefined };
+  }
+
+  const prevPage = pageList[currentPageNum - 1] ?? null;
+  const nextPage = pageList[currentPageNum + 1] ?? null;
+
+  return { prevPage, nextPage };
+}
+
 export default function AppLayoutDocsFooter() {
   const classes = useStyles();
   const t = useTranslate();
   const userLanguage = useUserLanguage();
-  const { activePage, pages } = React.useContext(PageContext);
+  const { activePage } = React.useContext(PageContext);
   const [rating, setRating] = React.useState();
   const [comment, setComment] = React.useState('');
   const [commentOpen, setCommentOpen] = React.useState(false);
   const [snackbarOpen, setSnackbarOpen] = React.useState(false);
   const [snackbarMessage, setSnackbarMessage] = React.useState(false);
   const inputRef = React.useRef();
-  const pageList = flattenPages(pages);
-  const currentPageNum = findIndex(pageList, (page) => page.pathname === activePage?.pathname);
-  const currentPage = pageList[currentPageNum];
```

Comment:
If `activePage` was not `null` then `currentPage === activePage` i.e. `currentPage` was redundant.